### DISCUSSION TITLE
Form not "prevent" when using calling submit via $refs

### DISCUSSION THREAD

**Discussion by andyto** [alpinejs/alpine#3846] [7. Bugs (archived)] [answered]
I have the following code and trying to do a ajax call when a user change the value from a select. However, it the form is not stopped. It worked when I use submit button instead.

```
<div x-data='{}'>
  <form @submit.prevent='console.log("do ajax")' x-ref='form'>
    <select @change='$refs.form.submit()'>
      <option value="1">1</option>
      <option value="2">2</option>
    </select>
    
    <input type='submit'>
  </form>
</div>
```

**Comment by SimoTod** [comment, marked as answer]
That is the expected behavior. Submitting a form in that way via js always submit the form skipping event and form validation: https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit

You need to use requestSubmit

**Reply by andyto** [reply]
Great. Thank you for the reference.

**Comment by SimoTod** [comment]
Also, you don't need x-ref. A form element  already knows which form it belongs to `$el.form.`

**Comment by ekwoka** [comment]
It isn't dispatching a submit event, so it won't get caught by the listener.
